crates/rpc/rpc-types/src/eth/trace/parity.rs

```diff
@@ -350,6 +349,70 @@ pub struct LocalizedTransactionTrace {
     pub transaction_position: Option<u64>,
 }
 
+impl Serialize for LocalizedTransactionTrace {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let mut s = serializer.serialize_struct("LocalizedTransactionTrace", 8)?;
+
+        match &self.trace.action {
+            Action::Call(_call_action) => {
+                s.serialize_field("action", _call_action)?;
+            }
+            Action::Create(_create_action) => {
+                s.serialize_field("action", _create_action)?;
+            }
+            Action::Selfdestruct(_selfdestruct_action) => {
+                s.serialize_field("action", _selfdestruct_action)?;
+            }
+            Action::Reward(_reward_action) => {
+                s.serialize_field("action", _reward_action)?;
+            }
+        }
+        if let Some(block_hash) = self.block_hash {
+            s.serialize_field("blockHash", &block_hash)?;
+        }
+        if let Some(block_number) = self.block_number {
+            s.serialize_field("blockNumber", &block_number)?;
+        }
+        if let Some(transaction_hash) = self.transaction_hash {
+            s.serialize_field("transactionHash", &transaction_hash)?;
+        }
+        if let Some(transaction_position) = self.transaction_position {
+            s.serialize_field("transactionPosition", &transaction_position)?;
+        }
+        match &self.trace.result {
+            Some(TraceOutput::Call(_call)) => {
+                s.serialize_field("result", _call)?;
+            }
+            Some(TraceOutput::Create(_create)) => {
+                s.serialize_field("result", _create)?;
+            }
+            None => {}
+        }
+        s.serialize_field("subtraces", &self.trace.subtraces)?;
+        s.serialize_field("traceAddress", &self.trace.trace_address)?;
+
+        match &self.trace.action {
+            Action::Call(_call_action) => {
+                s.serialize_field("type", "call")?;
+            }
+            Action::Create(_create_action) => {
+                s.serialize_field("type", "create")?;
+            }
+            Action::Selfdestruct(_selfdestruct_action) => {
+                s.serialize_field("type", "selfdestruct")?;
+            }
+            Action::Reward(_reward_action) => {
+                s.serialize_field("type", "reward")?;
+            }
+        }
```

Comment:
I'd like to add an enum for this, like `enum ActionKind` and then add a function to action `fn kind(&self) -> ActionKind`